examples/partial_labels.py

```diff
@@ -0,0 +1,182 @@
+"""
+TODO How to revert to previous model (in the same model group) if necessary?
+
+Outside of Script
+* Expose API or script for comparing model metrics between original model and PL model
+    * Determine if we should add new value matching metrics into somewhere more "official"
+    than the solutions toolkit
+* Optionally provide script for comparing metrics between workflows, but this
+    might not be required for the first pass
+* Determine one vs many
+    * Can prepopulate based on statistics and then ask client to correct
+* Need to move model from DEV to PROD after training to create a prod workflow
+    * https://indicodata.slack.com/archives/CLJGMCK3Q/p1626895835321200
+"""
+import argparse
+from datetime import datetime
+
+from indico import IndicoClient, IndicoConfig
+from indico_toolkit.staggered_loop import StaggeredLoop
+
+
+def create_argument_parser() -> argparse.ArgumentParser:
+    """
+    Argument parser for training a donor model
+    """
+    parser = argparse.ArgumentParser(description="Train a partial labels model")
+    parser.add_argument(
+        "--dev_host",
+        required=True,
+        help="Host URL for DEV Indico IPA environment",
+    )
+    parser.add_argument(
+        "--dev_api_token_path",
+        required=True,
+        help="Path to DEV API token to authenticate to dev_host",
+    )
+    parser.add_argument(
+        "--prod_host",
+        required=True,
+        help="Host URL for PROD Indico IPA environment",
+    )
+    parser.add_argument(
+        "--prod_api_token_path",
+        required=True,
+        help="Path to PROD API token to authenticate to prod_host",
+    )
+    parser.add_argument(
+        "--dev_dataset_id",
+        required=True,
+        help="Dataset ID for dev dataset associated w/ prod workflow",
+    )
+    parser.add_argument(
+        "--dev_questionnaire_id",
+        required=True,
+        help="Questionnaire (teach task) ID associated with dev dataset",
+    )
+    parser.add_argument(
+        "--dev_model_group_id",
+        required=True,
+        help="Model Group ID for dev dataset / teach task",
+    )
+    parser.add_argument(
+        "--dev_workflow_id",
+        required=True,
+        help="Workflow ID for dev workflow associated with teach task",
+    )
+    parser.add_argument(
+        "--prod_workflow_id",
+        required=True,
+        help="Workflow ID for prod workflow with review data",
+    )
+    parser.add_argument(
+        "--max_review_docs",
+        default=1000,
+        help="Maximum number of documents from review to add to dataset",
+    )
+    parser.add_argument(
+        "--oversample_errs",
+        action="store_true",
+        help="Sample more heavily from documents with more post review corrections",
+    )
+    return parser
+
+
+def main(
+    dev_host: str,
+    dev_api_token_path: str,
+    prod_host: str,
+    prod_api_token_path: str,
+    dev_dataset_id: int,
+    dev_questionnaire_id: int,
```

Comment:
General thought: organizationally, we probably need to rebrand 'questionnaire_id' to 'teach_id' and do a better job of meaningfully exposing this in the platform (I also wonder why questionnaire_id is necessary given model_group_id should be enough to backward fill questionnaire_id- 1-to-1 relationship)